From acc057e84d10c8f71f474fa6d25402544d16b30e Mon Sep 17 00:00:00 2001
From: "henry@bluestacks.com" <henry@bluestacks.com>
Date: Wed, 6 Dec 2023 15:35:50 +0800
Subject: [PATCH 099/224] [A13] add preinstalled for SYSTEM_ALERT_WINDOW

add preinstalled for SYSTEM_ALERT_WINDOW

com.bluestacks.home uses SAW, grant them SAW through preinstalled
---
 core/res/AndroidManifest.xml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/res/AndroidManifest.xml b/core/res/AndroidManifest.xml
index 6a80d1cb62a7..8ec47c185895 100644
--- a/core/res/AndroidManifest.xml
+++ b/core/res/AndroidManifest.xml
@@ -3196,7 +3196,7 @@
     <permission android:name="android.permission.SYSTEM_ALERT_WINDOW"
         android:label="@string/permlab_systemAlertWindow"
         android:description="@string/permdesc_systemAlertWindow"
-        android:protectionLevel="signature|setup|appop|installer|pre23|development" />
+        android:protectionLevel="signature|preinstalled|setup|appop|installer|pre23|development" />
 
     <!-- @SystemApi @hide Allows an application to create windows using the type
          {@link android.view.WindowManager.LayoutParams#TYPE_APPLICATION_OVERLAY},
